### ISSUE TITLE
Cannot compile NSI script

### ISSUE THREAD

**Issue by lychichem** [idleberg/vscode-nsis#24]
VS code version: 1.54.1 x64 system install with zh-CN language pack
Windows Version: 10.0.19043.841 x64 Pro (zh-CN)
NSIS version: 3.06.1
extension version: 3.39.0
Script: https://gist.github.com/lychichem/d63254461b8de65e6f458101036686e5
VS code web developer Console log:

```
workbench.desktop.main.js:62 [Extension Host] Using makensis path found in user settings: C:\Users\leich\Downloads\Tools\NSIS\makensis.exe
workbench.desktop.main.js:62 [Extension Host] makensis: cmdHelp() has been deprecated and will be removed in future versions, please use commandHelp() instead
v @ workbench.desktop.main.js:62
$logExtensionHostMessage @ workbench.desktop.main.js:1687
_doInvokeHandler @ workbench.desktop.main.js:1687
_invokeHandler @ workbench.desktop.main.js:1687
_receiveRequest @ workbench.desktop.main.js:1687
_receiveOneMessage @ workbench.desktop.main.js:1687
(anonymous) @ workbench.desktop.main.js:1687
fire @ workbench.desktop.main.js:57
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:57
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
I @ workbench.desktop.main.js:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:62 [Extension Host] rejected promise not handled within 1 second: TypeError: Cannot read property '_languageId' of undefined
v @ workbench.desktop.main.js:62
$logExtensionHostMessage @ workbench.desktop.main.js:1687
_doInvokeHandler @ workbench.desktop.main.js:1687
_invokeHandler @ workbench.desktop.main.js:1687
_receiveRequest @ workbench.desktop.main.js:1687
_receiveOneMessage @ workbench.desktop.main.js:1687
(anonymous) @ workbench.desktop.main.js:1687
fire @ workbench.desktop.main.js:57
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:57
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
I @ workbench.desktop.main.js:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:62 [Extension Host] stack trace: TypeError: Cannot read property '_languageId' of undefined	at Nn (c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:3647)	at c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:7377	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21605	at Object.edit (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:49569)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21592	at S._executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:103774)	at S.$executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:104212)	at u._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10509)	at u._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10201)	at u._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:8871)	at u._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:7673)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:5782	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:29119	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)	at t._receiveMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:20693)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:17587	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)	at l.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12808)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12156	at Socket.E (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:12375)	at Socket.emit (events.js:315:20)	at addChunk (_stream_readable.js:295:12)	at readableAddChunk (_stream_readable.js:271:9)	at Socket.Readable.push (_stream_readable.js:212:10)	at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
v @ workbench.desktop.main.js:62
$logExtensionHostMessage @ workbench.desktop.main.js:1687
_doInvokeHandler @ workbench.desktop.main.js:1687
_invokeHandler @ workbench.desktop.main.js:1687
_receiveRequest @ workbench.desktop.main.js:1687
_receiveOneMessage @ workbench.desktop.main.js:1687
(anonymous) @ workbench.desktop.main.js:1687
fire @ workbench.desktop.main.js:57
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:57
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
I @ workbench.desktop.main.js:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:2084 [[object Object]]Cannot read property '_languageId' of undefined
$onExtensionRuntimeError @ workbench.desktop.main.js:2084
_doInvokeHandler @ workbench.desktop.main.js:1687
_invokeHandler @ workbench.desktop.main.js:1687
_receiveRequest @ workbench.desktop.main.js:1687
_receiveOneMessage @ workbench.desktop.main.js:1687
(anonymous) @ workbench.desktop.main.js:1687
fire @ workbench.desktop.main.js:57
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:57
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
I @ workbench.desktop.main.js:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:2084 TypeError: Cannot read property '_languageId' of undefined
	at Nn (c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:3647)
	at c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:7377
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21605
	at Object.edit (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:49569)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21592
	at S._executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:103774)
	at S.$executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:104212)
	at u._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10509)
	at u._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10201)
	at u._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:8871)
	at u._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:7673)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:5782
	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:29119
	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
	at t._receiveMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:20693)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:17587
	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
	at l.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12808)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12156
	at Socket.E (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:12375)
	at Socket.emit (events.js:315:20)
	at addChunk (_stream_readable.js:295:12)
	at readableAddChunk (_stream_readable.js:271:9)
	at Socket.Readable.push (_stream_readable.js:212:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
$onExtensionRuntimeError @ workbench.desktop.main.js:2084
_doInvokeHandler @ workbench.desktop.main.js:1687
_invokeHandler @ workbench.desktop.main.js:1687
_receiveRequest @ workbench.desktop.main.js:1687
_receiveOneMessage @ workbench.desktop.main.js:1687
(anonymous) @ workbench.desktop.main.js:1687
fire @ workbench.desktop.main.js:57
fire @ workbench.desktop.main.js:76
_receiveMessage @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
fire @ workbench.desktop.main.js:57
acceptChunk @ workbench.desktop.main.js:76
(anonymous) @ workbench.desktop.main.js:76
I @ workbench.desktop.main.js:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
```

**Comment by idleberg** [comment]
Can you reproduce the error with a Minimal, Reproducible Example. For obvious reasons, I cannot compile the example script.

Also, did you get an error notification such as "This command is only available for NSIS files"?

**Comment by lychichem** [comment]
@idleberg I created a sample repo (https://github.com/lychichem/nsis-bug), which I cannot compile inside vscode but can compile in makensisw from nsis distribution. The log in console is this:
workbench.desktop.main.js:sourcemap:62 [Extension Host] rejected promise not handled within 1 second: TypeError: Cannot read property '_languageId' of undefined
v @ workbench.desktop.main.js:sourcemap:62
$logExtensionHostMessage @ workbench.desktop.main.js:sourcemap:1687
_doInvokeHandler @ workbench.desktop.main.js:sourcemap:1687
_invokeHandler @ workbench.desktop.main.js:sourcemap:1687
_receiveRequest @ workbench.desktop.main.js:sourcemap:1687
_receiveOneMessage @ workbench.desktop.main.js:sourcemap:1687
(anonymous) @ workbench.desktop.main.js:sourcemap:1687
fire @ workbench.desktop.main.js:sourcemap:57
fire @ workbench.desktop.main.js:sourcemap:76
_receiveMessage @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
fire @ workbench.desktop.main.js:sourcemap:57
acceptChunk @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
I @ workbench.desktop.main.js:sourcemap:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:sourcemap:62 [Extension Host] stack trace: TypeError: Cannot read property '_languageId' of undefined	at Nn (c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:3647)	at c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:7377	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21605	at Object.edit (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:49569)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21592	at S._executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:103774)	at S.$executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:104212)	at u._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10509)	at u._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10201)	at u._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:8871)	at u._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:7673)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:5782	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:29119	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)	at t._receiveMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:20693)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:17587	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)	at l.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12808)	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12156	at Socket.E (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:12375)	at Socket.emit (events.js:315:20)	at addChunk (_stream_readable.js:295:12)	at readableAddChunk (_stream_readable.js:271:9)	at Socket.Readable.push (_stream_readable.js:212:10)	at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
v @ workbench.desktop.main.js:sourcemap:62
$logExtensionHostMessage @ workbench.desktop.main.js:sourcemap:1687
_doInvokeHandler @ workbench.desktop.main.js:sourcemap:1687
_invokeHandler @ workbench.desktop.main.js:sourcemap:1687
_receiveRequest @ workbench.desktop.main.js:sourcemap:1687
_receiveOneMessage @ workbench.desktop.main.js:sourcemap:1687
(anonymous) @ workbench.desktop.main.js:sourcemap:1687
fire @ workbench.desktop.main.js:sourcemap:57
fire @ workbench.desktop.main.js:sourcemap:76
_receiveMessage @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
fire @ workbench.desktop.main.js:sourcemap:57
acceptChunk @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
I @ workbench.desktop.main.js:sourcemap:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:sourcemap:2084 [[object Object]]Cannot read property '_languageId' of undefined
$onExtensionRuntimeError @ workbench.desktop.main.js:sourcemap:2084
_doInvokeHandler @ workbench.desktop.main.js:sourcemap:1687
_invokeHandler @ workbench.desktop.main.js:sourcemap:1687
_receiveRequest @ workbench.desktop.main.js:sourcemap:1687
_receiveOneMessage @ workbench.desktop.main.js:sourcemap:1687
(anonymous) @ workbench.desktop.main.js:sourcemap:1687
fire @ workbench.desktop.main.js:sourcemap:57
fire @ workbench.desktop.main.js:sourcemap:76
_receiveMessage @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
fire @ workbench.desktop.main.js:sourcemap:57
acceptChunk @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
I @ workbench.desktop.main.js:sourcemap:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
workbench.desktop.main.js:sourcemap:2084 TypeError: Cannot read property '_languageId' of undefined
	at Nn (c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:3647)
	at c:\Users\leich\.vscode\extensions\idleberg.nsis-3.39.0\lib\extension.js:22:7377
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21605
	at Object.edit (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:49569)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:21592
	at S._executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:103774)
	at S.$executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:104212)
	at u._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10509)
	at u._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10201)
	at u._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:8871)
	at u._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:7673)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:5782
	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:29119
	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
	at t._receiveMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:20693)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:17587
	at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
	at l.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12808)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12156
	at Socket.E (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:12375)
	at Socket.emit (events.js:315:20)
	at addChunk (_stream_readable.js:295:12)
	at readableAddChunk (_stream_readable.js:271:9)
	at Socket.Readable.push (_stream_readable.js:212:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
$onExtensionRuntimeError @ workbench.desktop.main.js:sourcemap:2084
_doInvokeHandler @ workbench.desktop.main.js:sourcemap:1687
_invokeHandler @ workbench.desktop.main.js:sourcemap:1687
_receiveRequest @ workbench.desktop.main.js:sourcemap:1687
_receiveOneMessage @ workbench.desktop.main.js:sourcemap:1687
(anonymous) @ workbench.desktop.main.js:sourcemap:1687
fire @ workbench.desktop.main.js:sourcemap:57
fire @ workbench.desktop.main.js:sourcemap:76
_receiveMessage @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
fire @ workbench.desktop.main.js:sourcemap:57
acceptChunk @ workbench.desktop.main.js:sourcemap:76
(anonymous) @ workbench.desktop.main.js:sourcemap:76
I @ workbench.desktop.main.js:sourcemap:2760
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186

**Comment by idleberg** [comment]
I'm afraid this error doesn't make any sense to me. Just to be certain:

1. You open the NSIS file inside of VSCode?
2. You run the *Save & Compile* command from either the command palette or the menu button?
3. The document has focus when you run the command?

I've added a debug message for testing and I cannot produce the case where `Cannot read property '_languageId' of undefined`. This property even exists in a non-editor view.

If you have some more patience in getting to the bottom of the problem, could you please try the following:

1. Install the [Extension API](https://marketplace.visualstudio.com/items?itemName=idleberg.extension-api) package
2. Open the NSIS file you tried compiling
3. Run the command *Extension API: window.activeTextEditor*
4. Does this output the properties `_documentData` / `_documentData._languageId` to the output window?

**Comment by lychichem** [comment]
> I'm afraid this error doesn't make any sense to me. Just to be certain:
> 
> 1. You open the NSIS file inside of VSCode?
> 2. You run the _Save & Compile_ command from either the command palette or the menu button?
> 3. The document has focus when you run the command?
> 
> I've added a debug message for testing and I cannot produce the case where `Cannot read property '_languageId' of undefined`. This property even exists in a non-editor view.
> 
> If you have some more patience in getting to the bottom of the problem, could you please try the following:
> 
> 1. Install the [Extension API](https://marketplace.visualstudio.com/items?itemName=idleberg.extension-api) package
> 2. Open the NSIS file you tried compiling
> 3. Run the command _Extension API: window.activeTextEditor_
> 4. Does this output the properties `_documentData` / `_documentData._languageId` to the output window?
> 
> The example script you posted works fine for me, so I don't think it's related to that.

Seems like vscode 1.54.1 breaks something. Reported to MS team.

**Comment by idleberg** [comment]
> Seems like vscode 1.54.1 breaks something. Reported to MS team.

That seems to be the reason. It should be possible to support both, older versions and 1.54. I will look into it this evening.

**Comment by idleberg** [comment]
Should be fixed v3.40 🤞

**Comment by lychichem** [comment]
> Should be fixed v3.40 🤞

Confirmed. Thanks for your high speed!
